# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: js/src/gc/Marking.cpp
# Commit: 0857304e4134
# Full Hash: 0857304e41340efdad42d8f1c78d0a0206491b5c
# Author: Jon Coppeard <jcoppeard@mozilla.com>
# Date: 2022-11-10 04:48:58
# Description:
#   Bug 1799678 - Fix assertion that assumed it could be only happen on the main thread r=sfink
#   
#   JS::RuntimeHeapIsMinorCollecting() gets the JSContext from TLS, but
#   IsMarkedInternal() can be called from a helper thread where this is null.
#   
# ==============================================================================

diff -r 50db218ab916 -r 0857304e4134 js/src/gc/Marking.cpp
--- a/js/src/gc/Marking.cpp	Wed Nov 09 09:24:14 2022 +0000
+++ b/js/src/gc/Marking.cpp	Wed Nov 09 09:37:00 2022 +0000
@@ -2130,7 +2130,7 @@
 bool js::gc::IsMarkedInternal(JSRuntime* rt, T* thing) {
   // Don't depend on the mark state of other cells during finalization.
   MOZ_ASSERT(!CurrentThreadIsGCFinalizing());
-  MOZ_ASSERT(!JS::RuntimeHeapIsMinorCollecting());
+  MOZ_ASSERT(rt->heapState() != JS::HeapState::MinorCollecting);
   MOZ_ASSERT(thing);
   CheckIsMarkedThing(thing);
 